src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,41 @@
 
 class Animal {
   // write your code here
+  static alive = [];
+
+  constructor(name, health) {
+    this.name = name;
+    this.health = health || 100;
+    Animal.alive.push(this);
+  }
+  static deadAnimal() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
   // write your code here
+  constructor(health, name, hidden) {
+    super(health, name);
+
+    this.hidden = hidden || false;
+  }
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  bite(herb) {
```

Comment:
the parameter name is misleading

a) you are not guaranteed to get `Herbivore`
b) `herb` is a plant, not an animal :)

```suggestion
  bite(animal) {
```

```suggestion
  bite(other) {
```